bin/basket.js

```diff
@@ -14,88 +14,98 @@ var config = require('../config').getProperties()
-        return
-      }
-      // Forward all others to basket API.
-      request(
-        {
-          url: config.basket.apiUrl + '/fxa-register/',
-          strictSSL: true,
-          method: 'POST',
-          headers: {
-            'X-API-Key': config.basket.apiKey
-          },
-          form: {
-            fxa_id: message.uid,
-            email: message.email,
-            // Basket won't accept empty or null `accept_lang` field,
-            // so we default to en-US.  This should only happen if
-            // the user has not sent an explicit Accept-Language header.
-            accept_lang: message.locale || 'en-US'
-          }
-        },
-        function (err, res, body) {
-          // Log and retry for network-level errors.
-          if (err) {
-            log.error(
-              {
-                op: 'basketRequest',
-                uid: message.uid,
-                locale: message.locale,
-                err: err
-              }
-            )
-            return
-          }
-          // Log at error level for HTTP-level errors, info level otherwise.
-          if (res.statusCode < 200 || res.statusCode >= 300) {
-            log.error(
-              {
-                op: 'basketRequest',
-                status: res.statusCode,
-                uid: message.uid,
-                locale: message.locale,
-                body: body
-              }
-            )
-          } else {
-            log.info(
-              {
-                op: 'basketRequest',
-                status: res.statusCode,
-                locale: message.locale,
-                body: body
-              }
-            )
-          }
-          message.del()
-        }
-      )
-    }
-    else {
-      message.del()
+
+    var messageHandler = messageHandlers[message.event]
+    if (messageHandler) {
+      return messageHandler(message)
     }
+
+    message.del()
   }
 )
 basketQueue.start()
+
+function onVerified (message) {
+  forwardEvent(message, '/fxa-register/', {
+    fxa_id: message.uid,
+    email: message.email,
+    // Basket won't accept empty or null `accept_lang` field,
+    // so we default to en-US.  This should only happen if
+    // the user has not sent an explicit Accept-Language header.
+    accept_lang: message.locale || 'en-US'
+  })
+}
+
+function onLogin (message) {
+  forwardEvent(message, '/fxa-activity/', {
+    activity: 'account.login',
+    service: message.service,
+    fxa_id: message.uid,
+    first_device: message.isFirstDevice,
+    user_agent: message.userAgent
+  })
+}
+
+function forwardEvent (message, endpoint, data) {
+  // Ignore email addresses that are clearly from dev testing.
+  if (shouldIgnoreEmail(message.email)) {
+    message.del()
+    return
+  }
+
+  // Forward all others to basket API.
+  request({
+    url: config.basket.apiUrl + endpoint,
+    strictSSL: true,
+    method: 'POST',
+    headers: {
+      'X-API-Key': config.basket.apiKey
+    },
+    form: data,
```

Comment:
Per comment, we'll need to use `json` here for activity and `form` for register, unless @pmclanahan is feel adventurous enough to push through a patch to accept json bodies everywhere :-)